src/filter_tree/filter_tree.c

```diff
@@ -166,6 +166,11 @@ int _applyFilter(SIValue *aVal, SIValue *bVal, AST_Operator op) {
 		return (op == OP_NEQUAL);
 	}
 
+	if(disjointOrNull == COMPARED_NAN) {
+		/* The filter passes if we're testing for inequality, and fails otherwise. */
+		return (op == OP_NEQUAL);
```

Comment:
@AviAvni  Missing test here, I guess it's pretty easy to test it.
Just a suggestion: I think you can merge this if with the one above.

Reply:
yes this is checked in the TCK tests

Reply:
which test missing?

Reply:
test added thanks